lib/hyrax/transactions/steps/save.rb

```diff
@@ -38,13 +38,23 @@ def call(change_set, user: nil)
             unsaved.respond_to?(:permission_manager)
 
           user ||= ::User.find_by_user_key(saved.depositor)
-          Hyrax.publisher.publish('object.deposited', object: saved, user: user) if unsaved.new_record
-          Hyrax.publisher.publish('object.metadata.updated', object: saved, user: user)
 
+          publish_changes(unsaved, saved, user)
           Success(saved)
         rescue StandardError => err
           Failure([err.message, change_set.resource])
         end
+
+        private
+
+        def publish_changes(unsaved, saved, user)
```

Comment:
seems good. i take it this means you plan to re-use this transaction step for the collection save?

at some point in that process, we should add unit tests that run it against a collection.

Reply:
This will be used in the controller to create a collection.  I have this working in my local branch and hope it will be ready to commit soon.